src/test/scala/mesosphere/marathon/core/instance/TestInstanceBuilder.scala

```diff
@@ -106,8 +107,21 @@ case class TestInstanceBuilder(
       host = hostName.getOrElse(instance.agentInfo.host),
       region = region.orElse(instance.agentInfo.region),
       zone = zone.orElse(instance.agentInfo.zone),
-      attributes = attributes.getOrElse(instance.agentInfo.attributes)
-    )))
+      attributes = attributes.getOrElse(instance.agentInfo.attributes))))
+
+  def withReservation(volumeIds: Seq[LocalVolumeId]): TestInstanceBuilder =
+    withReservation(volumeIds, reservationStateNew)
+
+  def withReservation(state: Reservation.State): TestInstanceBuilder =
+    withReservation(Seq.empty, state)
+
+  def withReservation(volumeIds: Seq[LocalVolumeId], state: Reservation.State): TestInstanceBuilder =
+    withReservation(Reservation(volumeIds, state))
+
+  def withReservation(reservation: Reservation): TestInstanceBuilder =
+    copy(instance = instance.copy(reservation = Some(reservation)))
+
+  def reservationStateNew = Reservation.State.New(timeout = None)
```

Comment:
the only usage of this field right now is 10 lines above so maybe we don't need this field anymore...

Reply:
Yes, it is on the roadmap already, because I would like to merge this PR sooner, so that we can work on finishing PVs for pods together in parallel.